File: spoj/tyama_spojXMAX_TLE.rb
#!/usr/bin/ruby
B=62
def solve(a)
	idx=0
	(B-1).downto(0){|i|
		maxI=idx
		maxE=-1
		(idx...a.size).each{|j|
			if a[j][i]>0 && a[j]>maxE
				maxE=a[j]
				maxI=j
			end
		}
		next if maxE<0
		a[idx],a[maxI]=a[maxI],a[idx]
		maxI=idx
		a.size.times{|j|
			if j!=maxI && a[j][i]>0
				a[j]=a[j]^a[maxI]
			end
		}
		idx+=1
	}
	a.reduce(:^)
end
n,*a=$<.read.split.map(&:to_i)
p solve(a)

File: spoj/tyama_spojXMAX_TLE.py
#!/usr/bin/python
import sys
if sys.version_info[0]<3: range=xrange

B=62
def solve(a):
	idx=0
	for i in range(B-1,-1,-1):
		maxI=idx
		maxE=-1
		for j in range(idx,len(a)):
			if a[j]&(1<<i) and a[j]>maxE:
				maxE=a[j]
				maxI=j
		if maxE<0: continue
		a[idx],a[maxI]=a[maxI],a[idx]
		maxI=idx
		for j in range(len(a)):
			if j!=maxI and a[j]&(1<<i):
				a[j]=a[j]^a[maxI]
		idx+=1
	r=0
	for e in a: r^=e
	return r

a=[int(e) for e in sys.stdin.read().split()]
print(solve(a[1:]))
